contracts/staking/contracts/src/stake/MixinStake.sol

```diff
@@ -17,188 +17,217 @@
 */
 
 pragma solidity ^0.5.9;
+pragma experimental ABIEncoderV2;
 
-import "@0x/contracts-utils/contracts/src/LibRichErrors.sol";
 import "@0x/contracts-utils/contracts/src/LibSafeMath.sol";
-import "../libs/LibStakingRichErrors.sol";
-import "../libs/LibRewardMath.sol";
 import "../immutable/MixinConstants.sol";
 import "../immutable/MixinStorage.sol";
 import "../interfaces/IStakingEvents.sol";
 import "./MixinZrxVault.sol";
 import "../staking_pools/MixinStakingPoolRewardVault.sol";
+import "../staking_pools/MixinStakingPoolRewards.sol";
 import "../sys/MixinScheduler.sol";
 import "./MixinStakeBalances.sol";
-import "./MixinTimeLockedStake.sol";
+import "./MixinStakeStorage.sol";
 
 
 /// @dev This mixin contains logic for managing ZRX tokens and Stake.
-/// Stake is minted when ZRX is deposited and burned when ZRX is withdrawn.
```

Comment:
That's a good question. I was also thinking about the best way to do this. The spec that I've written so far is in markdown. I'm thinking that we'll just put it right here in the repo. And then it can have indices that map to the code. So if you're looking at a mixin like 🤔 ~ then you'd hit up the spec and search for the Mixin. We'll mention this in the README too.